src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs

```diff
@@ -78,10 +78,6 @@ private static string GetTenantId(IAzureContext context)
             if (context.Account == null)
                 throw new ArgumentException(KeyVaultProperties.Resources.ArmAccountNotFound);
 
-            if (context.Account.Type != AzureAccount.AccountType.User &&
-                context.Account.Type != AzureAccount.AccountType.ServicePrincipal)
-                throw new ArgumentException(string.Format(KeyVaultProperties.Resources.UnsupportedAccountType, context.Account.Type));
-
             if (context.Subscription != null && context.Account != null)
                 tenantId = context.Subscription.GetPropertyAsArray(AzureSubscription.Property.Tenants)
                        .Intersect(context.Account.GetPropertyAsArray(AzureAccount.Property.Tenants))
```

Comment:
@markcowl This code retrieves looks for the tenant ID in two different places.  The code around this line is intersecting subscription tenants with account tenants.  If we don't find the tenant here, then we retrieve the tenant from the context Tenant property.  Do you know of a reason that we shouldn't just retrieve the tenant directly from the context rather than doing the whole intersecting thing?

Also, are we guaranteed to be able to get a tenant id regardless of Account.Type?  In other words, can we/should we throw if we don't find a tenant id at all?

Reply:
@RandalliLama It is best practice to get the tenant from the context